File: seminar/src/main/java/com/server/dosopt/seminar/common/exception/GlobalExceptionHandler.java
package com.server.dosopt.seminar.common.exception;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
public class GlobalExceptionHandler {

    @ExceptionHandler(IllegalArgumentException.class)
    public ResponseEntity<Void> handleIllegalArgumemtException(final IllegalArgumentException e) {
        return ResponseEntity.badRequest().build();   // 400 응답으로 반환
    }

    /**
     * 커스텀 Exception에 대한 예외처리 (한꺼번에 처리 가능)
     */
    @ExceptionHandler(BusinessException.class)
    public ResponseEntity<Void> handleBusinessException(BusinessException e) {
        return ResponseEntity.badRequest().build();
    }
}

File: kotlin-seminar/src/main/kotlin/sopt/org/kotlinseminar/common/exception/GlobalExceptionHandler.kt
package sopt.org.kotlinseminar.common.exception

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.MethodArgumentNotValidException
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice


@RestControllerAdvice
class GlobalExceptionHandler {

    @ExceptionHandler(IllegalArgumentException::class)
    fun handleIllegalArgumentException(e: java.lang.IllegalArgumentException): ResponseEntity<ErrorResponse> {
        return ResponseEntity.badRequest().body(ErrorResponse(e.message))

    }

    /**
     * 커스텀 Exception에 대한 예외처리 (한꺼번에 처리 가능)
     */
    @ExceptionHandler(BusinessException::class)
    fun handleBusinessException(e: BusinessException): ResponseEntity<Void> {
        return ResponseEntity.badRequest().build()
    }
}